jerry-core/include/jerryscript-compiler.h

```diff
@@ -41,8 +41,10 @@ extern "C"
 #define JERRY_ATTR_PURE __attribute__((pure))
 #define JERRY_ATTR_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
 
+#ifndef _CPP_CHECK_
```

Comment:
This macro was used to prevent `JERRY_LIKELY(x)` to use `__builtin_expect` since it's only a GCC extension and cppcheck does not know how to interpret it correctly. Until this workaround there were false positive alarms for the following like of code parts:

```c
any_kind *obj_p = do_somenthing();

if (JERRY_UNLIKELY (obj_p == NULL)) {
 return;
}

JERRY_ASSERT(obj_p);
obj_p->anything++; // warning(nullPointerRedundantCheck): Either the condition 'obj_p==NULL' is redundant or there is possible null pointer dereference: obj_p.
```

So now whenever the cppcheck is running the JERRY_{UN}LIKELY macros are forced to `JERRY_{UN}LIKELY(x)`, without `_CPP_CHECK_`. See [here](https://github.com/jerryscript-project/jerryscript/pull/4515/files?file-filters%5B%5D=.sh&file-filters%5B%5D=No+extension#diff-99756c8cc2d02b40a4a5b9f55999a2c07243a412738886847e8600c8e4808c24R40).

Moreover, there are no new inline suppressions by the patch and the number of already existing ones have been reduced to 1. This can be resolved later, but I didn't find a good solution it for now.